### ISSUE TITLE
Error handling CustomChoices with similar text representations

### ISSUE THREAD

**Issue by coderazzi** [coderazzi/tablefilter-swing#26]
(Original issue 26 created by None on 2012-09-02T12:18:55+00:00)

```
What steps will reproduce the problem?
1. I add two CustomChoices. One has the textual representation 'Cob'; the second, is
'Cob - 1'
2- I select the CustomChoice with name 'Cob', that should match all dates in the column
with date=yesterday

What is the expected output? What do you see instead?

The rows with the given date should not be filtered out. As it happens, all rows gets
filtered out.

What version of the product are you using? On what operating system?

4.4.0 on Windows XP

Please provide any additional information below.


```

Original issue reported on code.google.com by `luismpena` on 2012-09-02 12:18:55

**Comment by coderazzi** [comment]
(Original issue 26 last updated on 2012-09-02T12:18:55+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by coderazzi** [comment]
(Original comment by coderazzi on 2012-09-02T14:08:04+00:00)

```
Fixed on version 4.4.1
```

Original issue reported on code.google.com by `coderazzi` on 2012-09-02 14:08:04

**Comment by coderazzi** [comment]
(Original comment by coderazzi on 2012-09-02T14:23:38+00:00)

```
As a note, the error would only happen if one CustomChoice' text representation was
a substring (from the beginning) of other CustomChoice's representation.
In this case, selecting the shorter CustomChoice would filter out all rows in the table.
```

Original issue reported on code.google.com by `coderazzi` on 2012-09-02 14:23:38
